core/services/directrequest/delegate.go

```diff
@@ -98,22 +121,37 @@ func (l *listener) Start() error {
 	if !connected {
 		return errors.New("Failed to register listener with logBroadcaster")
 	}
-	l.unsubscribeLogs = unsubscribe
+	go l.run()
+
+	hbUnsubscribe := l.headBroadcaster.Subscribe(l)
+
+	l.shutdownWaitGroup.Add(1)
+	go func() {
+		defer hbUnsubscribe()
```

Comment:
Maybe rename to `unsubscribeHeads` while you're at it?

Reply:
👍🏼